spring-kafka/src/main/java/org/springframework/kafka/listener/MessageListener.java

```diff
@@ -26,8 +30,12 @@
  *
  * @author Marius Bogoevici
  * @author Gary Russell
+ * @author Sanghyeok An
  */
 @FunctionalInterface
 public interface MessageListener<K, V> extends GenericMessageListener<ConsumerRecord<K, V>> {
 
+	default void setCallbackForAsyncFailureQueue(Consumer<FailedRecordTuple<K, V>> asyncRetryCallback) {
```

Comment:
Can we hide this behind `java.util.function.BiConsumer` abstraction?

I may see a value in the `FailedRecordTuple`, but let's try to push it down to the `KafkaMessageListenerContainer` internals!

Reply:
IMHO, we need to use `FailedRecordTuple`. 
Otherwise, it's hard to manage the `ConsumerRecord` and the `Exception` queues.!

As you mentioned, I changed the code to use `BiConsumer` and removed dependency on `FailedRecordTuple` , except in `KafkaMessageListenerContainer`. 
Now, `FailedRecordTuple` has become an internal record.

Does this make sense to you?
Please let me know!